### ISSUE TITLE
Range operator with invalid LHS errors

### ISSUE THREAD

**Issue by schas002** [cheddar-lang/Cheddar#6]
**Version:** 0.0.317-alpha

A code snippet is worth a thousand words:

```
cheddar> :|                                                                                                               
readline.js:925                                                                                                           
            throw err;                                                                                                    
            ^                                                                                                             
                                                                                                                          
TypeError: Cannot read property 'constructor' of undefined                                                                
    at CheddarEval.step (/usr/local/share/cheddar/dist/interpreter/core/eval/eval.js:148:34)                              
    at CheddarEval.exec (/usr/local/share/cheddar/dist/interpreter/core/eval/eval.js:249:34)                              
    at CheddarExec.step (/usr/local/share/cheddar/dist/interpreter/exec.js:39:29)                                         
    at CheddarExec.exec (/usr/local/share/cheddar/dist/interpreter/exec.js:54:22)                                         
    at Interface.<anonymous> (/usr/local/share/cheddar/dist/cli/repl.js:60:24)                                            
    at emitOne (events.js:77:13)                                                                                          
    at Interface.emit (events.js:169:7)                                                                                   
    at Interface._onLine (readline.js:211:10)                                                                             
    at Interface._line (readline.js:550:8)                                                                                
    at Interface._ttyWrite (readline.js:827:14)
```

:|

**Comment by vihanb** [comment]
10/10 detailed description. Include the version next time pls

**Comment by schas002** [comment]
`:)` errors too, with the same behavior.

**Comment by vihanb** [comment]
I don't think the `(` and `)` are parsed as they are assumed to be the next statement. The tokenizer doesn't pick up on this so it tries to execute it, but it has no idea what to do with `:` because it's not an operator at the moment, or a variable, or a literal

**Comment by The-Quill** [comment]
I just ran into this as well.... apparently `2:4` is supposed to work, but string conversions are borked or something.... `String::(2:4)`

**Comment by vihanb** [comment]
Yeah I've refactored so the cast is given from the RHS. But I think LHS is still the owner of some casts. I'll fix in a bit

**Comment by vihanb** [comment]
This must of been due to my circular dependency fix. That operator behavior is defined in the number class but it requires the array class to construct I have removed this dependency which is why it's returning undefined :/ Will fix in 0.1.1a

**Comment by schas002** [comment]
\o/
